src/js/jsx/tools/GuidesOverlay.jsx

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React),
+        StoreWatchMixin = Fluxxor.StoreWatchMixin,
+        d3 = require("d3"),
+        _ = require("lodash");
+
+    var OS = require("adapter/os");
+
+    var mathUtil = require("js/util/math");
+
+    // Used for debouncing the overlay drawing
+    var DEBOUNCE_DELAY = 200;
+
+    var GuidesOverlay = React.createClass({
+        mixins: [FluxMixin, StoreWatchMixin("document", "tool", "application", "ui")],
+
+        /**
+         * Keeps track of current mouse position so we can rerender the overlaid layers correctly
+         * @type {number}
+         */
+        _currentMouseX: null,
```

Comment:
This may just be a re-hashing of old conversations, but my immediate reaction when I see a bunch of private props as the opening salvo of a React component class is: why aren't these just tracked in the component state?